js/src/client.ts

```diff
@@ -1597,23 +1599,23 @@ export class Client implements LangSmithTracingClientInterface {
     context: string,
     options?: { apiKey?: string; apiUrl?: string }
   ) {
-    try {
-      // Create multipart form data boundary
-      const boundary =
-        "----LangSmithFormBoundary" + Math.random().toString(36).slice(2);
-      const body = await (_globalFetchImplementationIsNodeFetch()
-        ? this._createNodeFetchBody(parts, boundary)
-        : this._createMultipartStream(parts, boundary));
+    // Create multipart form data boundary
+    const boundary =
+      "----LangSmithFormBoundary" + Math.random().toString(36).slice(2);
 
+    const isNodeFetch = _globalFetchImplementationIsNodeFetch();
+    const buildBuffered = () => this._createNodeFetchBody(parts, boundary);
+    const buildStream = () => this._createMultipartStream(parts, boundary);
+
+    const send = async (body: BodyInit) => {
       const headers: Record<string, string> = {
         ...this.headers,
         "Content-Type": `multipart/form-data; boundary=${boundary}`,
       };
       if (options?.apiKey !== undefined) {
         headers["x-api-key"] = options.apiKey;
       }
-
-      const res = await this.batchIngestCaller.call(
+      return await this.batchIngestCaller.call(
```

Comment:
nit for the future - `return await` is unnecessary here:

https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise